src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,153 @@
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Logging properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Logging")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Services properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Services")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Configuration properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Configuration")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                            diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, IdentifierNameSyntax identifierMethod)
+    {
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        // builder.Host.ConfigureLogging(builder => builder.AddJsonConsole());
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); 
+
+        if (diagnosticTarget is InvocationExpressionSyntax invocation)
+        {
+            // No modification are made if the invocation isn't accessing a method on `builder.Host` or `builder.WebHost`.
+            if (invocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
+                || hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax configureMethodOnHostAccessExpr)
+            {
+                return document;
+            }
```

Comment:
It's generally not a good idea to return document without modification if you already registered an action. This means the lightbulb will have an item that says "Fix references to Configuration properties on WebApplicationBuilder", but it will do nothing. These kind of checks should be done *before* registering an action.